app/main.py

```diff
@@ -1,13 +1,20 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": str(error["code"]),
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [format_linter_error(error) for error in errors],
+            "path": file_path,
+            "status": ("failed" if errors else "passed")}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [format_single_linter_file(path, errors)
+            for path, errors in zip(linter_report, linter_report.values())]
```

Comment:
For improved readability and maintainability, it's generally recommended to iterate directly over the `items()` method of a dictionary to obtain both the `keys` and `values`, rather than using `zip`.